srcDoc/examples.rst

```diff
@@ -600,3 +600,344 @@ Maximum Throughput -- 2 *Routing Service*
+
+1. Copy the IDL files into srcIdl/custom/
+
+2. Implement the functions of customtype.cxx
+
+- **initialize_custom_type**:
+    This function is used to initialize your data.
+    The function takes one argument:
+
+        - A reference to the customer type.
+
+::
+
+    bool initialize_custom_type(RTI_CUSTOM_TYPE & data)
+    {
+        bool success = true;
+        if (! data.test_seq.test_seq.maximum(0)) {
+            success = false;
+        }
+        if (! data.test_seq.test_seq.ensure_length(SIZE_TEST_SEQ, SIZE_TEST_SEQ)) {
+            success = false;
+        }
+        data.test_enum = ENUM1;
+        return success;
+    }
+
+- **register_custom_type**:
+    This function is used to set your data before been register.
+    The function takes two arguments:
+
+        - A reference to the customer type.
+        - A specific number unique for every key.
+
+::
+
+    void register_custom_type(RTI_CUSTOM_TYPE & data, unsigned long key)
+    {
+        data.test_long = key;
+    }
+
+- **set_custom_type**:
+    This function is used to set your data before being sent.
+    The function takes three arguments:
+
+        - A reference to the customer type.
+        - A specific number unique for every key.
+        - The target size set by the command line parameter ``-dataLen <bytes>``
+
+::
+
+    bool set_custom_type(
+            RTI_CUSTOM_TYPE & data,
+            unsigned long key,
+            int target_data_len)
+    {
+        bool success = true;
+        data.test_long = key;
+        if (sprintf(data.test_string.test_string, "Hello World! %lu", key) < 0) {
+            success = false;
+        }
+        return success;
+    }
+
+- **finalize_data_custom_type**:
+    This function is used to remove your data. It is called in the destructor.
+    The function takes one argument:
+
+        - A reference to the customer type.
+
+::
+
+    bool finalize_data_custom_type(RTI_CUSTOM_TYPE & data)
+    {
+        return true;
+    }
+
+- **initialize_custom_type_dynamic**:
+    This function is used to initialize your DynamicData.
+    The function takes one argument:
+
+            - A reference to the full DDS_DynamicData.
+
+::
+
+    bool initialize_custom_type_dynamic(DDS_DynamicData & data)
+    {
+        try {
+            test_seq = new long[SIZE_TEST_SEQ];
+        } catch (std::bad_alloc& ba) {
+            fprintf(stderr, "bad_alloc test_seq  %s failed.\n",  ba.what());
+            return false;
+        }
+        bool success = long_seq.from_array(
+                (DDS_Long *) test_seq,
+                SIZE_TEST_SEQ);
+        if (! success) {
+            fprintf(stderr, "from_array(test_seq) failed.\n");
+            return false;
+        }
+        return true;
+    }
+
+- **register_custom_type_dynamic**:
+    This function is used to set your DynamicData before been register.
+    The function takes two arguments:
+
+        - A reference to the full DDS_DynamicData.
```

Comment:
DDS_DynamicData object (change it in all the occurrences).

Reply:
Done